internal/app/ui/cpwsarea/wsmap/pmap/psettings/screenshot.go

```diff
@@ -57,12 +61,28 @@ func (p *Panel) showScreenshot() {
 
 func (p *Panel) createScreenshot() {
 	p.sessionScreenshot.saving = true
-
-	width, height := p.editor.Dmm().MaxX*dmmap.WorldIconSize, p.editor.Dmm().MaxY*dmmap.WorldIconSize
+	toolSelect := tools.SetSelected(tools.TNGrab).(*tools.ToolGrab)
+	bounds := toolSelect.Bounds()
+	var width, height int
+	if p.sessionScreenshot.inselection && toolSelect.HasSelectedArea() {
+		width, height = (int(bounds.X2-bounds.X1)+1)*dmmap.WorldIconSize, (int(bounds.Y2-bounds.Y1)+1)*dmmap.WorldIconSize
+	} else if p.sessionScreenshot.inselection && !toolSelect.HasSelectedArea() {
+		appdialog.Open(appdialog.TypeInformation{
+			Title:       "Nothing selected!",
+			Information: "Screenshot in Selection is on, but you have nothing selected.",
+		})
+		p.sessionScreenshot.saving = false
+		return
+	} else {
+		width, height = p.editor.Dmm().MaxX*dmmap.WorldIconSize, p.editor.Dmm().MaxY*dmmap.WorldIconSize
+	}
 
 	c := canvas.New()
 	c.ClearColor = canvas.Color{} // Empty clear color with no alpha
 	c.Render().Camera.Level = p.editor.ActiveLevel()
+	if p.sessionScreenshot.inselection {
+		c.Render().Camera.Translate(-float32((int(bounds.X1)-1)*dmmap.WorldIconSize), -float32((int(bounds.Y1)-1)*dmmap.WorldIconSize))
+	}
```

Comment:
Optional recommendation, you can create bound vars at the start of the method with `0, 0` value. And then, if user is in selection mode modify them. And here you won't need to check `inselection` mode and just do 

```go
c.Render().Camera.Translate(boundX, boundY)
```

The logic flow will look a little bit cleaner.

Reply:
addressed, possibly

Reply:
addressed